chore: Readability and comprehensibility of code

Tessg22 opened the pull request:
- Add dry_run argument
- Rename variables
- Put if statement on line 120 into function for better comprehensibility of code

Tessg22 committed:
chore: Readability and comprehensibility of code

- Add dry_run argument
- Rename variables
- Put if statement on line 120 into function for better understanding

Tessg22 committed:
chore: Fix name of variable

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -84,7 +85,7 @@ def get_max_allowed_size(es, percentage_threshold):
         i = i + int(node)
 
-    max_allowed_size = (percentage_threshold * i) / 100.0
+    max_allowed_size = int( (percentage_threshold * i) / 100.0 )
```
Why should this be int?
I don't remember the functionality but in some languages converting 78.5 to int returns 78
If this is intentional ingore

Tessg22 replied:
On bellow line I had previously _limit=int(max_allowed_size)_ for the reason that units are already in bytes. And I am deleting based on bytes. Imho there is no reason for displaying max_allowed_size as float. Just round the max_allowed_size.

`log(log_info, "Do not add into actionable list: '{index}', summed disk usage is {usage} B and disk limit is {limit} B".format(
                index=index.name, usage=indices_size_counter, limit=max_allowed_size))`

I would still like to know your opinion on this Ron.

georgettica replied:
I stand corrected, and thanks for the elaborate response.
This should be in bytes and shouldn't be a float

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
     Returns a list of actionable indices based on percentage size and age.
     """
-    class data_structure:
+    class indices_struct:
```
Would rename to index but not critical

Tessg22 replied:
Done

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
-                indice=data_object.name, usage=size_counter, limit=int(max_allowed_size)))
-            size_counter += data_object.size
+    indices_size_counter = 0
+    for index in sorted(indices, key=lambda x: x.creation_date, reverse=True):
```
Should maybe add a comment on the reason for reverse true

Tessg22 replied:
Done.

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
+    for index in sorted(indices, key=lambda x: x.creation_date, reverse=True):
+        expected_size = index.size + indices_size_counter
+        def index_smaller_then_max_allowed_size(indices_size_counter, expected_size, max_allowed_size):
+            if indices_size_counter < max_allowed_size and expected_size < max_allowed_size:
```
I would just return the conditions in the if, it makes the code smaller and handles the false case aswell

georgettica replied:
Also maybe move the function somewhere else

Tessg22 replied:
Could you please review code if its the way you suggested now?

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -98,33 +99,40 @@ def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
+            if indices_size_counter < max_allowed_size and expected_size < max_allowed_size:
+                return True
+        if index_smaller_then_max_allowed_size(indices_size_counter, expected_size, max_allowed_size):
+            log(log_info, "Do not add into actionable list: '{index}', summed disk usage is {usage} B and disk limit is {limit} B".format(
```
Might I suggest using f strings if you are using an updated python version

https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/

Tessg22 replied:
Done.

georgettica reviewed (commented):
Have some stuff I found here, home my input helps

drewandersonnz commented on scripts/lightweight_curator.py:
```diff
@@ -158,7 +167,7 @@ def main():
     es = es_connect_args(elasticsearch_host)
 
-    log(log_info, "Removing indices which are above {percentage} threshold from host '{host}'".format(
+    log(log_info, "Searching for indices which are above {percentage}% threshold from host '{host}'".format(
```
This states you are searching for indices which themselves are individually above threshold. 

Maybe say something like "Removing/Searching for oldest indices to remove first when total storage exceeds threshold".

Tessg22 replied:
I took inspiration from your suggestion.
Could you please check if it looks good now?

Tessg22 committed:
chore: Minor code review changes

- Rename variables
- Change log message
- Move if-else statement into index_smaller_then_max_allowed_size func

Tessg22 committed:
chore: Fix indice typo

drewandersonnz reviewed (approved):
my request has been done

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -94,37 +95,49 @@ def get_first_item(a_dict={}):
     first_value = next(value_iterator)
     return first_value
 
+def index_smaller_then_max_allowed_size(index, limit, indices_size_counter, indices_to_delete):
```
If this returns a list then imo it shou be plural
I know it's very confusing where to put which term but you get the hang of it after a while
```suggestion
def indices_smaller_then_max_allowed_size(index, limit, indices_size_counter, indices_to_delete):
```

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -94,37 +95,49 @@ def get_first_item(a_dict={}):
+
+    return indices_to_delete
+
 def get_actionable_indices(es, max_allowed_size, index_name_prefixes_list):
```
```suggestion
def get_actionable_indices(es, max_allowed_size, index_name_prefixes):
```

georgettica commented on scripts/lightweight_curator.py:
```diff
@@ -94,37 +95,49 @@ def get_first_item(a_dict={}):
+    Appends index into the list of indices with their name, size and creation_date.
+    """
+    indices = []
     for index_name_prefixes in index_name_prefixes_list:
```
```suggestion
    for index_name_prefix in index_name_prefixes:
```

georgettica reviewed (commented):
Very tiny nits.

No need for second pass from my side after you fix them.

Can be approved and fixed later if you want this PR merged sooner rather than later

Tessg22 committed:
chore: fix prefixes/prefix and index/indices typo
